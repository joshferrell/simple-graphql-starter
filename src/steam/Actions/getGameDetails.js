import fetch from 'node-fetch';
import config from '../config/index';

const getGameDetails = gameId =>
    new Promise(async (resolve, reject) => {
        const url = `${config.steam.url}/ISteamUserStats/GetSchemaForGame/v2/?key=${config.steam.apiKey}&appid=${gameId}`;

        try {
            const res = await fetch(url);

            if (res.status !== 200) {
                throw new Error('Response from server not successful');
            }

            const body = await res.json();
            resolve(body.game.gameName);
        } catch (e) {
            console.log(e);
            reject('Unable to access steam api');
        }
    });

export default getGameDetails;
