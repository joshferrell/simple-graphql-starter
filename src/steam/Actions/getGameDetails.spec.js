import getGameDetails from './getGameDetails';

describe('get steam game details', () => {
    it('should return game information', async () => {
        const game = await getGameDetails('218620');
        expect(game).toMatchSnapshot();
    });

    it('should reject if the game does not exist', () =>
        expect(getGameDetails('a')).rejects
    );
});
