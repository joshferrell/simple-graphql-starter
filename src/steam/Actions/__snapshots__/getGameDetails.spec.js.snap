// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get steam game details should return game information 1`] = `"PAYDAY 2"`;
